sql-to-dbsp-compiler/lib/sqllib/src/array.rs

```diff
@@ -259,3 +260,95 @@ pub fn sequence__(start: i32, end: i32) -> Vec<i32> {
 }
 
 some_function2!(sequence, i32, i32, Vec<i32>);
+
+// translated from the Calcite implementation to match the behavior
+pub fn arrays_overlapNvec_Nvec_<T>(first: Vec<Option<T>>, second: Vec<Option<T>>) -> Option<bool>
+where
+    T: Eq + Hash,
+{
+    if first.len() > second.len() {
+        return arrays_overlapNvec_Nvec_(second, first);
+    }
+
+    let (smaller, bigger) = (first, second);
+    let mut has_null = false;
+
+    if !smaller.is_empty() && !bigger.is_empty() {
+        let mut shset: HashSet<Option<T>> = HashSet::from_iter(smaller);
+        has_null = shset.remove(&None);
```

Comment:
you should not scan the array at all if this is true

Reply:
I might be misunderstanding your suggestion but I don't think that works. 

We still need to compare if both the arrays contain the same element. 

Consider: `ARRAYS_OVERLAP(ARRAY [2, null], ARRAY [1, 2, 3, null])`
Here `has_null` would be true. 
But just based on this, we cannot return `NULL` or `true`.

We still need to check if they have any overlapping elements, in this case 2.

Reply:
So if they do not have elements in common but any of them has a null the result is null? Maybe you can write exactly this. This is a really bizarre function definition. For most of these functions it like like nulls are an afterthought.